File: AtBash.cpp
#include <iostream>

using namespace std;

void cipherEncryption(){
    string message;
    cout << "Enter Message: ";
    getline(cin, message);
    cin.ignore();

    //message to upper case
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    string alpa = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string reverseAlpa = "";
    for(int i = alpa.length()-1; i > -1; i--){
        reverseAlpa += alpa[i];
    }

    string encryText = "";
    for(int i = 0; i < message.length(); i++){
        if(message[i] == 32){
            encryText += " ";
        } else {
            for(int j = 0; j < alpa.length(); j++){
                if(message[i] == alpa[j]){
                    encryText += reverseAlpa[j];
                    break;
                }
            } // inner for
        } // if-else
    } // for

    cout << "Encrypted Text: " << encryText;
}

void cipherDecryption(){
    string message;
    cout << "Enter Encrypted Message: ";
    getline(cin, message);
    cin.ignore();

    //message to upper case
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    string alpa = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string reverseAlpa = "";
    for(int i = alpa.length()-1; i > -1; i--){
        reverseAlpa += alpa[i];
    }

    string dencryText = "";
    for(int i = 0; i < message.length(); i++){
        if(message[i] == 32){
            dencryText += " ";
        } else {
            for(int j = 0; j < reverseAlpa.length(); j++){
                if(message[i] == reverseAlpa[j]){
                    dencryText += alpa[j];
                    break;
                }
            } // inner for
        } // if-else
    } // for

    cout << "Decrypted Text: " << dencryText;
}

int main()
{
    int choice;
    cout << "1. Encryption\n2. Decryption\nChoose(1,2): ";
    cin >> choice;
    cin.ignore();

    if(choice == 1){
        cout << "Encryption" << endl;
        cipherEncryption();
    } else if (choice == 2){
        cout << "Decryption" << endl;
        cipherDecryption();
    } else {
        cout << "Wrong Choice" << endl;
    }
    return 0;
}

File: Atbash.py
def at_encryption():
    alpa = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    # reversing alphabets of alpa variable
    rev_alpa = alpa[::-1]

    message = input("Enter message: ").upper();

    encry_text = ""

    for i in range(len(message)):
        if message[i] == chr(32):
            encry_text += " "
        else:
            for j in range(len(alpa)):
                if message[i] == alpa[j]:
                    encry_text += rev_alpa[j]
                    break
                # if
            # inner for
        # if-else
    # for

    print("Encrypted Text: {}".format(encry_text))


def at_decryption():
    alpa = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    # reversing alphabets of alpa variable
    rev_alpa = alpa[::-1]

    message = input("Enter message: ").upper();

    dencry_text = ""

    for i in range(len(message)):
        if message[i] == chr(32):
            dencry_text += " "
        else:
            for j in range(len(rev_alpa)):
                if message[i] == rev_alpa[j]:
                    dencry_text += alpa[j]
                    break
                    # if
                    # inner for
                    # if-else
    # for

    print("Decrypted Text: {}".format(dencry_text))


def main():
    choice = int(input("1. Encryption\n2.Decryption\nChoose(1,2): "))
    if choice == 1:
        print("---Encryption---")
        at_encryption()
    elif choice == 2:
        print("---Decryption---")
        at_decryption()
    else:
        print("Wrong Choice")


if __name__ == "__main__":
    main()
